semgrep-core/src/engine/Match_search_mode.ml

```diff
@@ -364,15 +364,21 @@ let rec filter_ranges env xs cond =
                    ( fk,
                      [
                        G.Arg (G.L (G.String (re_str, fk)) |> G.e);
-                       G.Arg (G.N (G.Id ((mvar, fk), fki)) |> G.e);
+                       G.Arg
+                         (G.Call
+                            ( G.N (G.Id (("str", fk), fki)) |> G.e,
+                              ( fk,
+                                [ G.Arg (G.N (G.Id ((mvar, fk), fki)) |> G.e) ],
+                                fk ) )
+                         |> G.e);
                      ],
                      fk ) )
                |> G.e
              in
 
              let env =
                if const_prop && (fst env.config).constant_propagation then
-                 Eval_generic.bindings_to_env_just_strings_const_prop bindings
+                 Eval_generic.bindings_to_env (fst env.config) bindings
```

Comment:
nice way to reuse str here so we can get rid of bindings_to_env_just_strings_const_prop!